cli/compose/types/types.go

```diff
@@ -77,69 +78,96 @@ type Config struct {
 	Configs  map[string]ConfigObjConfig
 }
 
+// MarshalYAML makes Config implement yaml.Marshaller
+func (c *Config) MarshalYAML() (interface{}, error) {
+	m := map[string]interface{}{}
+	if c.Services != nil {
+		services := map[string]ServiceConfig{}
+		for _, service := range c.Services {
+			s := service
+			s.Name = ""
+			services[service.Name] = s
+		}
+		m["services"] = services
+	}
+	if c.Networks != nil {
```

Comment:
Are these `nil` checks necessary (below)? If it is nil, it can assign nil without any problems, right? I think they can be removed.

Reply:
Right 😛